tests/rados/test_noscrub_nodeepscrub_check.py

```diff
@@ -0,0 +1,218 @@
+        # Modified the scrub parameters
+        mon_object.set_config(section="osd", name="osd_scrub_min_interval", value=60)
+        mon_object.set_config(section="osd", name="osd_deep_scrub_interval", value=420)
+        mon_object.set_config(section="osd", name="debug_osd", value="20/20")
+
+        # Check for log
+        exec_end_time = datetime.datetime.now() + datetime.timedelta(seconds=20)
+        while exec_end_time > datetime.datetime.now():
+            end_time, _ = installer.exec_command(
+                cmd="sudo date '+%Y-%m-%dT%H:%M:%S.%3N+0000'"
+            )
+            end_time = end_time.strip()
+            for pg_id in pg_id_list:
+                log_lines = [
+                    rf"{pg_id}\s*(?!deep-)scrub starts",
+                    rf"{pg_id}\s*deep-scrub starts",
+                ]
+                if verify_scrub_log(
+                    rados_object, init_time, end_time, pg_id, log_lines
+                ):
+                    log.error(
+                        f"The scrub started on the {pg_id} after setting the no_scrub and no_deepscrub flag"
+                    )
+                    return 1
+            time.sleep(30)
+
+        log.info(
+            "Verify that the deep-scrub starts after unsetting the nodeep-scrub flag"
+        )
+        init_time, _ = installer.exec_command(
+            cmd="sudo date '+%Y-%m-%dT%H:%M:%S.%3N+0000'"
+        )
+        init_time = init_time.strip()
+        scrub_obj.set_osd_flags("unset", "nodeep-scrub")
+        log.info("The nodeep-scrub flag is unset")
+        time.sleep(20)
+        deep_scrub_falg = False
+        exec_end_time = datetime.datetime.now() + datetime.timedelta(seconds=90)
+        while exec_end_time > datetime.datetime.now():
+            end_time, _ = installer.exec_command(
+                cmd="sudo date '+%Y-%m-%dT%H:%M:%S.%3N+0000'"
+            )
+            end_time = end_time.strip()
+            for pg_id in pg_id_list:
+                log_lines = [
+                    rf"{pg_id}\s*deep-scrub starts",
+                    rf"{pg_id}\s*deep-scrub ok",
+                ]
+                if verify_scrub_log(
+                    rados_object, init_time, end_time, pg_id, log_lines
+                ):
+                    log.info(
+                        f"The scrub started on the {pg_id} after unsetting no_deepscrub flag"
+                    )
+                    deep_scrub_falg = True
+            time.sleep(30)
+        if not deep_scrub_falg:
+            log.error(
+                "The deep-scrub not started after unsetting the nodeep-scrub flag"
+            )
+            return 1
+
+        # unset the paramters
+        mon_object.remove_config(section="osd", name="osd_scrub_min_interval")
+        mon_object.remove_config(section="osd", name="osd_deep_scrub_interval")
+        log.info("Wait for the cluster into active+clean state")
+        method_should_succeed(wait_for_clean_pg_sets, rados_object, timeout=3600)
```

Comment:
wait_for_clean_pg_sets method ignores scrub & deep-scrub states while checking for PG states. 

Since your workflow can fail if there are any PGs in scrubbing state already, can you create a new method to make sure that there are no active scrubs/deep-scrubs going on? just like wait_for_clean_pg_sets, but for scrub operations.

Reply:
wait_for_clean_pg_sets if fine, in the further steps I am checking the scrub or deep-scrub states in the logs.
I created a method to wait for the scrub/deep scrub to be completed in the cluster. IMO that is not required.